File: server/zwei-iot-device/src/main/java/com/zwei/iot/device/domain/Device.java
package com.zwei.iot.device.domain;

import com.zwei.common.core.domain.BaseEntity;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.SuperBuilder;

import java.io.Serial;

/**
 * 设备表 device
 *
 * @author zwei
 */
@Setter
@Getter
@SuperBuilder
@NoArgsConstructor
@AllArgsConstructor
public class Device extends BaseEntity {
    @Serial
    private static final long serialVersionUID = 1L;

    /**
     * 主键ID
     */
    private Long id;

    /**
     * 设备编号
     */
    private String code;

    /**
     * 设备名称
     */
    private String name;

    /**
     * 设备SN
     */
    private String sn;

    /**
     * 设备类型:0单参数,1多参数,2本地组网
     */
    private Integer deviceType;

    /**
     * 网络类型:0蜂窝,1NB-Iot
     */
    private Integer networkType;

    /**
     * 接入协议:MQTT/HTTP/COAP
     */
    private String protocolType;

    /**
     * 注册来源:MANUAL/API/IMPORT
     */
    private String registerSource;

    /**
     * 厂商名称
     */
    private String vendorName;

    /**
     * 设备接入用户名
     */
    private String authUsername;

    /**
     * 设备接入密码
     */
    private String authPassword;

    /**
     * 账号状态:1有效,2禁用
     */
    private Integer authStatus;

    /**
     * 设备图标
     */
    private String icon;

    /**
     * 图标路径
     */
    private String iconPath;

    /**
     * 业务状态（人工维护）: 1-正常, 2-维修, 3-停用
     */
    private Integer status;

    /**
     * 状态名称（查询时返回）
     */
    private String statusName;

    /**
     * 实时在线状态（JOIN device_online_status 表）: 1-在线, 0/null-离线
     */
    private Integer onlineStatus;

    /**
     * 最近上报时间
     */
    private String lastReportTime;

    /**
     * 注册时间
     */
    private String registeredAt;

    /**
     * 最近鉴权时间
     */
    private String lastAuthTime;

    /**
     * 最近鉴权IP
     */
    private String lastAuthIp;

    /**
     * 经度
     */
    private java.math.BigDecimal longitude;

    /**
     * 纬度
     */
    private java.math.BigDecimal latitude;

    /**
     * 删除标记: 0-正常, 1-删除
     */
    private Integer delFlag;

    /**
     * 传感器数量（列表查询时由子查询填充）
     */
    private Integer sensorCount;

    /**
     * 传感器列表（查询详情时返回）
     */
    private Object sensors;

    /**
     * 关联的隐患点ID（业务规则：1设备≤1隐患点，Service层富化）
     */
    private Long boundHazardPointId;

    /**
     * 关联的隐患点名称（Service层富化）
     */
    private String boundHazardPointName;

    /**
     * 隐患点ID（查询参数，用于筛选绑定到指定隐患点的设备）
     */
    private Long hazardPointId;

    @Override
    public String toString() {
        return "Device{" +
                "id=" + id +
                ", code='" + code + '\'' +
                ", name='" + name + '\'' +
                ", sn='" + sn + '\'' +
                ", deviceType=" + deviceType +
                ", networkType=" + networkType +
                ", protocolType='" + protocolType + '\'' +
                ", registerSource='" + registerSource + '\'' +
                ", vendorName='" + vendorName + '\'' +
                ", authUsername='" + authUsername + '\'' +
                ", authStatus=" + authStatus +
                ", icon='" + icon + '\'' +
                ", iconPath='" + iconPath + '\'' +
                ", status=" + status +
                ", statusName='" + statusName + '\'' +
                ", lastReportTime='" + lastReportTime + '\'' +
                ", registeredAt='" + registeredAt + '\'' +
                ", lastAuthTime='" + lastAuthTime + '\'' +
                ", lastAuthIp='" + lastAuthIp + '\'' +
                ", delFlag=" + delFlag +
                '}';
    }
}

File: web/src/api/device.ts
import request from '@/utils/request'
import type {AjaxResult, PageResult} from './system'

export interface DeviceItem {
  id?: number
  code: string
  name: string
  sn?: string
  deviceType?: number | null
  networkType?: number | null
  protocolType?: string
  registerSource?: string
  vendorName?: string
  authUsername?: string
  authPassword?: string
  authStatus?: number
  icon?: string
  iconPath?: string
  status: number
  statusName?: string
    onlineStatus?: number
  lastReportTime?: string
  registeredAt?: string
  lastAuthTime?: string
  lastAuthIp?: string
  longitude?: number | null
  latitude?: number | null
    /** 关联的隐患点 ID(用于地图选点时叠加其边界) */
    boundHazardPointId?: number | null
  createTime?: string
  sensors?: any[]
  /**
   * 传感器数量
   * - 列表接口当前不返回该字段，UI 层以占位符 - 展示
   * - 后端在 Device 实体中已具备 sensors 字段，分页查询可补一个子查询以填充 sensorCount
   */
  sensorCount?: number
}

export interface DeviceCreatePayload {
  code: string
  name: string
  sn?: string
  deviceType?: number | null
  networkType?: number | null
  protocolType?: string
  vendorName?: string
  icon?: string
  iconPath?: string
  longitude?: number | null
  latitude?: number | null
  status: number
  boundHazardPointId?: number | null
}

export interface DeviceUpdatePayload {
  name: string
  sn?: string
  deviceType?: number | null
  networkType?: number | null
  protocolType?: string
  vendorName?: string
  icon?: string
  iconPath?: string
  longitude?: number | null
  latitude?: number | null
  status: number
  boundHazardPointId?: number | null
}

export interface DeviceAuthAccount {
  deviceId: number
  username: string
  password: string
  authStatus: number
  registeredAt?: string
  lastAuthTime?: string
  lastAuthIp?: string
}

export interface DeviceCreateResult {
  id: number
  username: string
  password: string
}

export interface DevicePageParams {
  pageNum: number
  pageSize: number
  code?: string
  name?: string
  sn?: string
  status?: number | ''
}

const unwrap = async <T>(promise: Promise<AjaxResult<T>>): Promise<T> => {
  const response = await promise
    if (response && typeof response.code === 'number' && response.code !== 200) {
        throw new Error(response.msg || '操作失败')
    }
  return response.data
}

export const getDevicePage = (params: DevicePageParams) =>
  unwrap<PageResult<DeviceItem>>(request.get('/devices/page', { params }))

export const getDeviceDetail = (id: number) =>
  unwrap<DeviceItem>(request.get(`/devices/${id}`))

export const createDevice = (payload: DeviceCreatePayload) =>
  unwrap<DeviceCreateResult>(request.post('/devices', payload))

export const updateDevice = (id: number, payload: DeviceUpdatePayload) =>
  unwrap<{ id: number }>(request.put(`/devices/${id}`, payload))

export const deleteDevice = (id: number) =>
  unwrap<null>(request.delete(`/devices/${id}`))

export interface DeviceCopyPayload {
  code: string
  name: string
}

export const copyDevice = (id: number, payload: DeviceCopyPayload) =>
  unwrap<number>(request.post(`/devices/${id}/copy`, payload))

export const exportDevices = (params?: Record<string, any>) =>
  request.raw.post('/devices/export', params || {}, { responseType: 'blob' })

export const getDeviceAuthAccount = (id: number) =>
  unwrap<DeviceAuthAccount>(request.get(`/devices/${id}/auth-account`))

export const resetDevicePassword = (id: number, reason?: string) =>
  unwrap<{ username: string; password: string }>(request.post(`/devices/${id}/auth-password/reset`, { reason }))

export const changeDeviceAuthStatus = (id: number, authStatus: number, reason?: string) =>
  unwrap<DeviceAuthAccount>(request.put(`/devices/${id}/auth-status`, { authStatus, reason }))
